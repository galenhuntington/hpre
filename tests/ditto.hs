str1 :: String
''  = "test1"

str2 :: String
str2 = str3 where
   str3 :: String
   '' = "test2"

nested :: Int -> Bool
''  0 = a where
   a :: Bool
   '' = True
''  _ = b where
   b :: Bool
   b = False

column :: Int
'' = a where a :: Int
             a = 4

