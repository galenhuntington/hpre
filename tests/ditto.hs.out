str1 :: String
str1{-#COLUMN 5#-}= "test1"

str2 :: String
str2 = str3 where
   str3 :: String
   str3{-#COLUMN 7#-}= "test2"

nested :: Int -> Bool
nested{-#COLUMN 5#-}0 = a where
   a :: Bool
   a{-#COLUMN 7#-}= True
nested{-#COLUMN 5#-}_ = b where
   b :: Bool
   b = False

column :: Int
column{-#COLUMN 4#-}= a where a :: Int
             a = 4


